kroxylicious-kms/src/main/java/io/kroxylicious/kms/service/KekId.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.service;
+
+/**
+ * A semantic wrapper around KMS specific key id type
+ */
+public interface KekId {
```

Comment:
I can see adding in KekId makes the InBandManager more readable and easier to relate to the concepts being implemented, but does switching to `<K> K getId(Class<K> keyType)` make it harder to verify that we've strung together a KMS with the right combination of `keyIdSerde`, `resolveAlias` and `unencryptedKeyId`? It seems funky to me that the serde has to know "this KMS's implementation supports getting UUID".

Reply:
Yeah I hear you. My rationale was the serde has to know the concrete type anyway because its sole job is to translate the concrete type to bytes.